digital_image_processing/filters/gaussian_filter.py

```diff
@@ -21,8 +22,7 @@ def gaussian_filter(image, k_size, sigma):
     # im2col, turn the k_size*k_size pixels into a row and np.vstack all rows
     image_array = zeros((dst_height * dst_width, k_size * k_size))
     row = 0
-    for i in range(0, dst_height):
-        for j in range(0, dst_width):
+    for i, j in product(range(dst_height), range(dst_height)): 
```

Comment:
The previous implementation used both width and height so it would work for images that are not square.  The new implementation only uses height so it assumes that all images are square.

Reply:
Would https://docs.scipy.org/doc/numpy/reference/generated/numpy.prod.html be even faster?

Reply:
https://docs.scipy.org/doc/numpy/reference/generated/numpy.prod.html  is the internal product, it doesn't have the same functionality as itertools.product